helix-term/src/commands.rs

```diff
@@ -1469,6 +1469,34 @@ fn append_mode(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+#[cfg(test)]
+#[test]
+fn test_expand_path() {
+    assert!(expand_path("~/src/helix").starts_with('/'));
+}
+
+// we have to return String because we allocate in here
+fn expand_path(path: &str) -> String {
```

Comment:
We already have: https://github.com/helix-editor/helix/blob/master/helix-core/src/path.rs

Reply:
Why is it not used then to open files specified with `:open`? I would consider opening a 'new file' when I ask to open `~/.config/helix/config.toml` to be broken behavior.

Reply:
Just an oversight, I think.

Reply:
Roger that. I've updated the commit to use instead `helix-core`'s tilde expansion, and reverted the `Cargo.toml` changes in `helix-term`.

I can also undo the `Cargo.lock` changes, but I'm actually unsure why that's included in the first place. As far as I know, that's generated dynamically, and should generally not be tracked by `git`.